server/src/main/scala/actors/FlightsStateActor.scala

```diff
@@ -18,33 +20,53 @@ import scalapb.GeneratedMessage
 import server.protobuf.messages.FlightsMessage.UniqueArrivalMessage
 import services.SDate
-import services.crunch.deskrecs.GetFlights
+import services.crunch.deskrecs.{FlightsRequest, GetFlights, GetStateForDateRange, GetStateForTerminalDateRange}
 
 import scala.concurrent.ExecutionContextExecutor
 import scala.concurrent.duration._
 import scala.language.postfixOps
 
 
-class FlightsStateActor(initialMaybeSnapshotInterval: Option[Int],
-                        initialSnapshotBytesThreshold: Int,
-                        name: String,
-                        val now: () => SDateLike,
-                        expireAfterMillis: Int) extends PersistentActor with RecoveryActorLike with PersistentDrtActor[FlightsWithSplits] {
-  override def persistenceId: String = name
+object FlightsStateActor {
+  def isNonLegacyRequest(pointInTime: SDateLike, legacyDataCutoff: SDateLike): Boolean =
+    pointInTime.millisSinceEpoch >= legacyDataCutoff.millisSinceEpoch
+
+  def tempPitActorProps(pointInTime: SDateLike,
+                        message: DateRangeLike,
+                        now: () => SDateLike,
+                        queues: Map[Terminal, Seq[Queue]],
+                        expireAfterMillis: Int,
+                        legacyDataCutoff: SDateLike): Props = {
+    if (isNonLegacyRequest(pointInTime, legacyDataCutoff))
+      Props(new FlightsStateReadActor(now, expireAfterMillis, pointInTime.millisSinceEpoch, queues, legacyDataCutoff))
+    else
+      Props(new CrunchStateReadActor(1000, pointInTime, expireAfterMillis, queues, message.from, message.to))
+  }
+}
+
+class FlightsStateActor(val now: () => SDateLike, expireAfterMillis: Int, queues: Map[Terminal, Seq[Queue]], legacyDataCutoff: SDateLike)
+  extends PersistentActor with RecoveryActorLike with PersistentDrtActor[FlightsWithSplits] {
 
-  override val maybeSnapshotInterval: Option[Int] = initialMaybeSnapshotInterval
-  override val snapshotBytesThreshold: Int = initialSnapshotBytesThreshold
+  import FlightsStateActor._
+
+  override def persistenceId: String = "flights-state-actor"
+
+  override val maybeSnapshotInterval: Option[Int] = Option(5000)
+  override val snapshotBytesThreshold: Int = Sizes.oneMegaByte
   override val recoveryStartMillis: MillisSinceEpoch = now().millisSinceEpoch
 
   implicit val ec: ExecutionContextExecutor = context.dispatcher
 
-  val log: Logger = LoggerFactory.getLogger(s"$name-$getClass")
+  val log: Logger = LoggerFactory.getLogger(getClass)
 
   var state: FlightsWithSplits = FlightsWithSplits.empty
 
   var flightMinutesBuffer: Set[MillisSinceEpoch] = Set[MillisSinceEpoch]()
   var crunchSourceIsReady: Boolean = true
-  var maybeCrunchActor: Option[ActorRef] = None
+  var maybeCrunchQueueActor: Option[ActorRef] = None
+
+  implicit val timeout: Timeout = new Timeout(30 seconds)
+  val killActor: ActorRef = context.system.actorOf(Props(new RequestAndTerminateActor()))//, "flights-state-kill-actor")
```

Comment:
some commented out stuff here.

Reply:
we should figure out why naming the actors causes the app to crash.

Reply:
Added to to do ticket